analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -2664,10 +2664,12 @@ void NullCoalesce_Useless(string a, string b, string c, string d)
             //         ~~~~~~
 
             //Combo/Fatality
-            ret = notNull ?? isNull;    // Noncompliant
-            ret = isNull ?? null;       // Noncompliant
-
-            ret = "Value" ?? a; // Noncompliant
+            ret = notNull ?? isNull;    // Noncompliant {{Change this expression which always evaluates to the same result.}}
+            //    ^^^^^^^
+            ret = isNull ?? null;       // Noncompliant {{Change this expression which always evaluates to the same result.}}
+            //    ^^^^^^
+            ret = "Value" ?? a; // Noncompliant {{Change this expression which always evaluates to the same result.}}
+            //    ^^^^^^^
```

Comment:
We need more tests with messages. Testing all the different combinations of messages and values.